docs/source/basic_tutorial.rst

```diff
@@ -0,0 +1,216 @@
+
+Its goal is to provide researchers:
+
+- a unified framework for training and testing dialog models
+- multi-task training over many datasets at once
+- seamless integration of Amazon Mechanical Turk for data collection and human evaluation
+
+
+Getting Started
+---------------
+
+Agents
+^^^^^^
+
+The most basic concept in ParlAI is an Agent.
+Agents can be humans, a simple bot which repeats back anything that it hears,
+your perfectly-tuned neural network, a dataset being read out,
+or anything else that might send messages or interact with its environment.
+
+Agents have two primary methods they need to define:
+
+.. code-block:: python
+
+    def observe(self, observation)
+    def act(self)
+
+
+``observe`` notifies the agent of an action taken by another agent.
+
+``act`` produces an action from the agent.
+
+Observations are structured as a python `dict` with the following fields:
+
+
+.. image:: _static/img/act-obs-dict.png
+    :width: 60 %
+
+
+All of these fields are technically optional, and each task is free to use them
+as appropriate.
+
+Teachers
+^^^^^^^^
+
+A Teacher is special type of agent. They also implement the ``act`` and ``observe``
+functions like any agent does, but they also keep track of metrics which they
+return via a ``report`` function, such as the number of questions they have posed
+or how many times those questions have been answered correctly.
+
+Datasets will implement a subclass of Teacher, providing functions which
+download the dataset from its source if necessary, read the file into the
+right format, and provide an example with each call to the teacher's ``act``
+function.
+
+Exchanges between a student Agent and a bAbI task Teacher might look like the following dicts:
+
+.. code-block:: python
+
+    Teacher: {
+        'text': 'Sam went to the kitchen\nPat gave Sam the milk\nWhere is the milk?',
+        'labels': ['kitchen'],
+        'label_candidates': ['hallway', 'kitchen', 'bathroom'],
+        'episode_done': False  # indicates next example will be related to this one
+    }
+    Student: {
+        'text': 'hallway'
+    }
+    Teacher: {
+        'text': 'Sam went to the hallway\nPat went to the bathroom\nWhere is the milk?',
+        'labels': ['hallway'],
+        'label_candidates': ['hallway', 'kitchen', 'bathroom'],
+        'episode_done': True
+    }
+    Student: {
+        'text': 'hallway'
+    }
+    Teacher: {
+        ... # starts next episode
+    }
+    ...
+
+Worlds
+^^^^^^
+
+Worlds define the environment in which agents interact with one another. Worlds
+must implement a ``parley`` method, which conducts one set of interactions with
+each call.
+
+A simple world included in ParlAI, which all of our currently included tasks use,
+is the ``DialogPartnerWorld``. DialogPartnerWorld is initialized with two agents,
+and with each call to ``parley``, one exchange is done between the agents, in
+the following manner:
+
+.. code-block:: python
+
+    query = teacher.act()
```

Comment:
right yeah I thought this was waaayyyy more clear and so made better sense for the tutorial